src/axom/slic/core/LogStream.hpp

```diff
@@ -102,7 +104,17 @@ class LogStream
                       bool filter_duplicates) = 0;
 
   /*!
-   * \brief Flushes the log stream. It's a NO-OP by default.
+   * \brief Flushes the log stream on current rank. It's a NO-OP by default.
+   *
+   * \note The intent of this method is to be overridden by concrete
+   *  implementations. This is primarily useful for applications running
+   *  in a distributed MPI environment.
```

Comment:
I'm not sure if this `note` needs to be there.

On the other hand, it might be good to explicitly call out the fact that this function is not collective.